src/nsj_rest_lib/service/service_base.py

```diff
@@ -773,75 +773,90 @@ def _group_by_override_data(self, dto_list):
+            if list_field.relation_key_field is not None:
+                relation_key_field = list_field.relation_key_field
 
-                    if self._dto_class.pk_field not in dto.__dict__:
-                        raise DTOListFieldConfigException(
-                            f"PK field not found in DTO: {self._dto_class}"
-                        )
+            # Mapeia valor da chave -> lista de DTOs que possuem esse valor
+            key_to_dtos = {}
+            for dto in dto_list:
+                relation_filter_value = getattr(dto, relation_key_field, None)
+                if relation_filter_value is not None:
+                    key_to_dtos.setdefault(relation_filter_value, []).append(dto)
+                else:
+                    setattr(dto, master_dto_attr, [])
 
-                    # Resolvendo a chave do relacionamento
-                    relation_key_field = self._dto_class.pk_field
-                    if list_field.relation_key_field is not None:
-                        relation_key_field = list_field.relation_key_field
+            if not key_to_dtos:
+                continue
 
-                    # Getting value to filter related list
-                    relation_filter_value = getattr(dto, relation_key_field)
-                    if relation_filter_value is None:
-                        # If None, there is no related objects. So, set empty list and continue.
-                        setattr(dto, master_dto_attr, [])
-                        continue
+            # Instancia o service
+            if list_field.service_name is not None:
+                service = self._injector_factory.get_service_by_name(
+                    list_field.service_name
+                )
+            else:
+                service = ServiceBase(
+                    self._injector_factory,
+                    DAOBase(
+                        self._injector_factory.db_adapter(),
+                        list_field.entity_type,
+                    ),
+                    list_field.dto_type,
+                    list_field.entity_type,
+                )
+
+            # Monta o filtro IN para buscar todos os relacionados de uma vez
+            filters = {
+                list_field.related_entity_field: ",".join([str(key) for key in key_to_dtos])
+            }
 
-                    # Making filter to relation
-                    filters = {list_field.related_entity_field: relation_filter_value}
+            # Campos de particionamento: se existirem, só faz sentido se todos os DTOs tiverem o mesmo valor
+            # (caso contrário, teria que quebrar em vários queries)
+            # Aqui, só trata se todos tiverem o mesmo valor para cada campo de partição
+            for field in self._dto_class.partition_fields:
+                if field in list_field.dto_type.partition_fields:
+                    partition_values = set(getattr(dto, field, None) for dto in dto_list)
```

Comment:
Os campos de particionamento são obrigatoriamente passados como filtto num HTTP GET. Logo, todos os DTOs da listagem a devolver (e seus relacionados), devem estar numa mesma partição de dados.

Assim, acho que esse for aqui  pode ser economizado...